### PR TITLE
TIMX-18-marc-transform-yet-more-fields

### PR DESCRIPTION
**Pull request by ehanson8**
#### How can a reviewer manually see the effects of these changes?
Run the transform locally using the alma-2022-10-11-full-extracted-records-to-index.xml file in the dev S3 bucket, e.g.:

```
pipenv run transform -i <S3 URL for the above file> -o alma-transformed-records.json -s alma
```

#### What are the relevant tickets?
* https://mitlibraries.atlassian.net/browse/TIMX-18

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Includes new or updated dependencies?
NO

### CODE REVIEW

File: tests/test_marc.py [hunk 1]
```diff
@@ -148,6 +148,33 @@ def test_marc_record_all_fields_transform_correctly():
...
+                value="0878426914. paperback. alkaline paper", kind="ISBN"
+            ),
+            timdex.Identifier(value="0033-0736", kind="ISSN"),
+            timdex.Identifier(value="0095-9014", kind="ISSN"),
+            timdex.Identifier(
+                value="10.1596/978-0-8213-7468-9. doi", kind="Other Identifier"
+            ),
+            timdex.Identifier(value="1234567890. score. sewn", kind="Other Identifier"),
+            timdex.Identifier(value="(OCoLC)1312285564", kind="OCLC Number"),
+            timdex.Identifier(value="(OCoLC)on1312285564", kind="OCLC Number"),
```
Comment (ehanson8) [inline tests/test_marc.py:164]: This repetition of OCLC identifiers is very common so we may want to consider a dedupe block

File: tests/test_marc.py [hunk 2]
```diff
@@ -158,12 +185,179 @@ def test_marc_record_all_fields_transform_correctly():
...
+            ),
+            timdex.Note(value=["Advance copy notice inserted"], kind="Local Note"),
+        ],
         numbering="-Bd. 148, 4 (dez. 1997). Began in 1902.",
         physical_description=(
             "484 p. : ill. ; 30 cm. + 1 CD-ROM (4 3/4 in.). 1 DVD-ROM "
             "(4 3/4 in.). 1 vocal score (248 p.) ; 31 cm."
         ),
         publication_frequency=["Six no. a year", "Three times a year"],
+        related_items=[
```
Comment (ehanson8) [inline tests/test_marc.py:290]: It seems very hard to make all the `related_item.description` attributes "pretty" so I opted for minimizing excess punctuation, let me know what you think

File: transmogrifier/sources/marc.py [hunk 3]
```diff
@@ -283,6 +430,65 @@ def get_optional_fields(self, xml: Tag) -> dict:
...
+            },
+            {
+                "tag": "785",
+                "subfields": "abcdghikmnorstuwxyz",
+                "relationship": "Subsequent Title",
+            },
+            {
+                "tag": "787",
+                "subfields": "abcdghikmnorstuwxyz",
+                "relationship": "Related Item",
```
Comment (hakbailey) [inline transmogrifier/sources/marc.py:462]: We should probably default to `relationship=Not specified` here like we do for our other sources, for consistency.

File: tests/test_marc.py (hunk 2 above)
Comment (hakbailey) [reply tests/test_marc.py:290]: We need to review all of these related item subfields anyway so I just made a new ticket for that in the backlog. This is fine for now.

File: tests/test_marc.py (hunk 1 above)
Comment (hakbailey) [reply tests/test_marc.py:164]: Yeah, maybe make a ticket for that or add it to https://mitlibraries.atlassian.net/browse/TIMX-122?

**Review by hakbailey — COMMENTED** [review]
See inline comments for a few consistency suggestions, otherwise this looks good.

File: tests/test_marc.py [hunk 4]
```diff
@@ -148,6 +148,33 @@ def test_marc_record_all_fields_transform_correctly():
...
+            timdex.Identifier(
+                value="0878426914. paperback. alkaline paper", kind="ISBN"
+            ),
+            timdex.Identifier(value="0033-0736", kind="ISSN"),
+            timdex.Identifier(value="0095-9014", kind="ISSN"),
+            timdex.Identifier(
+                value="10.1596/978-0-8213-7468-9. doi", kind="Other Identifier"
+            ),
+            timdex.Identifier(value="1234567890. score. sewn", kind="Other Identifier"),
+            timdex.Identifier(value="(OCoLC)1312285564", kind="OCLC Number"),
```
[the remark this answers is not in the document]
Comment (ehanson8) [reply tests/test_marc.py:163]: Can I also lump this work in with [TIMX-122](https://mitlibraries.atlassian.net/browse/TIMX-122)?

[the remark this answers is not in the document]
Comment (hakbailey) [reply tests/test_marc.py:163]: That ticket will require some significant code changes, but removing the value string prefix is a one-line fix...any particular reason not to go ahead and do it?

[the remark this answers is not in the document]
Comment (ehanson8) [reply tests/test_marc.py:163]: I was thinking about the more general prefix removal, but I can just remove that particular (OCoLC) prefix

[the remark this answers is not in the document]
Comment (ehanson8) [reply tests/test_marc.py:163]: 👍

**Comment by ehanson8** [comment]
Pushed new commit

**Review by hakbailey — APPROVED** [review]
👍

**Pull request merged**